tempesta_fw/connection.c

```diff
@@ -90,13 +91,22 @@ tfw_connection_release(TfwConnection *conn)
  * for the life of @conn instance. The socket itself may have been
  * closed, but not deleted. ss_send() makes sure that data is sent
  * only on an active socket.
- *
- * @unref_data is true if we won't use @msg any more.
  */
 int
-tfw_connection_send(TfwConnection *conn, TfwMsg *msg, bool unref_data)
+tfw_connection_send(TfwConnection *conn, TfwMsg *msg)
 {
-	return ss_send(conn->sk, &msg->skb_list, unref_data);
+	int flags = 0;
+	const TfwHttpMsg *hm = (TfwHttpMsg *)msg;
+
+	if (TFW_CONN_TYPE(conn) & Conn_Clnt) {
+		if (tfw_http_msg_conn(hm) == TFW_HTTP_CONN_CLOSE)
+			flags |= SS_SEND_F_CONN_CLOSE;
```

Comment:
Yeah, agree. At least we don't suppose the behavior for now. However, I still think this is not the right place to set the flags, see https://github.com/tempesta-tech/tempesta/pull/574/files#r69903646